barretenberg/cpp/src/barretenberg/circuit_checker/circuit_checker.cpp

```diff
@@ -34,34 +34,90 @@ template <typename Builder> bool CircuitChecker::check(const Builder& builder_in
     TagCheckData tag_data;
     MemoryCheckData memory_data{ builder };
 
+    bool result = true;
+    // TODO(https://github.com/AztecProtocol/barretenberg/issues/870): Currently we check all relations for each block.
+    // Once sorting is complete, is will be sufficient to check only the relevant relation(s) per block.
+    size_t block_idx = 0;
```

Comment:
Previously I was only checking the `main` block (which contained all gates). Now I'm looping over all blocks and checking all relations on each one.